frontend/src/metabase/models/components/ModelDetailPage/ModelActionDetails/ModelActionDetails.tsx

```diff
@@ -0,0 +1,179 @@
+  ] = useToggle();
+
+  const canWrite = model.canWrite();
+  const hasImplicitActions = actions.some(action => action.type === "implicit");
+
+  const actionsSorted = useMemo(
+    () => _.sortBy(actions, mostRecentFirst),
+    [actions],
+  );
+
+  const menuItems = useMemo(() => {
+    return [
+      {
+        title: t`Create basic actions`,
+        icon: "bolt",
+        action: handleEnableImplicitActions,
+      },
+    ];
+  }, [handleEnableImplicitActions]);
+
+  const handleEditAction = useCallback(
+    (action: WritebackAction) => {
+      setEditingActionId(action.id);
+      showActionCreator();
+    },
+    [showActionCreator],
+  );
+
+  const handleCloseActionCreator = useCallback(() => {
+    hideActionCreator();
+    setEditingActionId(undefined);
+  }, [hideActionCreator]);
+
+  const renderActionListItem = useCallback(
+    (action: WritebackAction) => {
+      const onEdit = canWrite ? () => handleEditAction(action) : undefined;
+      return (
+        <li key={action.id} aria-label={action.name}>
+          <ModelActionListItem action={action} onEdit={onEdit} />
+        </li>
+      );
+    },
+    [canWrite, handleEditAction],
+  );
+
+  return (
+    <Root>
+      {canWrite && (
+        <ActionsHeader>
+          <Button onClick={showActionCreator}>{t`New action`}</Button>
+          {!hasImplicitActions && (
+            <ActionMenu
+              triggerIcon="ellipsis"
+              items={menuItems}
+              triggerProps={ACTION_MENU_TRIGGER_PROPS}
+            />
+          )}
+        </ActionsHeader>
+      )}
+      {actions.length > 0 ? (
+        <ActionList aria-label={t`Action list`}>
+          {actionsSorted.map(renderActionListItem)}
+        </ActionList>
+      ) : (
+        <NoActionsState
+          hasCreateButton={canWrite}
+          onCreateClick={handleEnableImplicitActions}
+        />
+      )}
+      {isActionCreatorOpen && (
+        <ActionCreator
+          modelId={model.id()}
+          databaseId={model.databaseId()}
+          actionId={editingActionId}
+          onClose={handleCloseActionCreator}
+        />
+      )}
+    </Root>
+  );
+}
+
+function NoActionsState({
+  hasCreateButton,
+  onCreateClick,
+}: {
+  hasCreateButton: boolean;
+  onCreateClick: () => void;
+}) {
+  return (
+    <EmptyState.Container>
+      <EmptyState.Title>{t`No actions have been created yet.`}</EmptyState.Title>
+      <EmptyState.Message>{t`Get started quickly with some basic actions to create, edit and delete, or create your own from scratch.`}</EmptyState.Message>
+      {hasCreateButton && (
+        <EmptyState.ActionContainer>
+          <Button
+            icon="bolt"
+            onClick={onCreateClick}
+          >{t`Create basic actions`}</Button>
+        </EmptyState.ActionContainer>
+      )}
+    </EmptyState.Container>
+  );
+}
+
+function mostRecentFirst(action: WritebackAction) {
+  const createdAt = new Date(action["created_at"]);
```

Comment:
We don't usually parse dates like this. Should we use `parseTimestamp` here? Or at least `moment-timezone`